packages/cli/test/unit/commands/logout/future.test.ts

```diff
@@ -0,0 +1,145 @@
+import { beforeEach, describe, expect, it, type MockInstance } from 'vitest';
+import { logout } from '../../../../src/commands/logout/future';
+import { client } from '../../../mocks/client';
+import { vi } from 'vitest';
+import fetch, { type Response } from 'node-fetch';
+import { as, VERCEL_CLI_CLIENT_ID } from '../../../../src/util/oauth';
+import { randomUUID } from 'node:crypto';
+import ua from '../../../../src/util/ua';
+
+const fetchMock = fetch as unknown as MockInstance<typeof fetch>;
+
+vi.mock('node-fetch', async () => ({
```

Comment:
We have a mock HTTP vercel server stored on the `client` (view a few other unit tests to get a sense for it) that would be preferable to use to avoid introducing a second alternate pattern to the tests.

Reply:
I first tried using `client`, but decided to use raw `fetch` instead because https://github.com/vercel/vercel/blob/main/packages/cli/src/util/client.ts/#L135-L221 had too much magic (adding/overriding headers, search params) and broke the requests/responses constantly.

We want to strictly follow the OAuth spec, so either I need to modify `client.fetch` via adding a `raw: true` flag to disable unexpected modifications, and refactor the `oauth.ts` file to rely on the `client.fetch`, or we keep the current mocks. Let me know what would be more preferable!